src/eos-updater-poll-lan.c

```diff
@@ -328,12 +320,8 @@ static guint
 parse_txt_version (const gchar *txt_version)
 {
   guint64 v;
-  const gchar *end = NULL;
-
-  errno = 0;
-  v = g_ascii_strtoull (txt_version, (gchar **)&end, 10);
 
-  if (errno != 0 || end == NULL || *end != '\0' || end == txt_version || v > G_MAXUINT)
+  if (!eos_string_to_unsigned (txt_version, 1, G_MAXUINT, &v, NULL))
```

Comment:
There is a comment just above this function (which github inconveniently hid):
`/* Valid version numbers start from 1. Return 0 on error. */`

Changing the lower bound to zero wouldn't change the behavior of the function, anyway.